docs/solutions/hypervisors/kvm/kvm-nested/README.md

```diff
@@ -0,0 +1,141 @@
+
+This sample code will be responsible to perform the following tasks:
+
+- Setup of all required network components: VCN, Internet Gateway, subnet, and security lists.
+
+- Spin up a [Virtual Machine](https://docs.us-phoenix-1.oraclecloud.com/Content/Compute/Concepts/computeoverview.htm) Instance based on Oracle Linux 7.4 image.
+
+- Attach [a secondary VNIC](https://docs.us-phoenix-1.oraclecloud.com/Content/Network/Tasks/managingVNICs.htm) to the VM instance.
+
+-	Installation of the KVM hypervisor provided by the Linux Kernel provided as a [Terraform module](https://www.terraform.io/docs/modules/usage.html) (reusable artifact). This process requires instance restart due to the kernel changes.
+
+-	Download of the image file.
+
+- Create a KVM domain for the guest VM.
+
+- Attach the secondary network interface (in PCI-passthrough mode) as network interfaces to the guest VM.
+
+- We don't cover the setup of the Guest VM, but you can connect to it via VNC or running `virsh connect <KVM Domain>` from the KVM host instance.
+
+
+Requirements
+------------
+
+- Access to OCI environment
+- [Generated OCI keys and OCIDs](https://docs.us-phoenix-1.oraclecloud.com/Content/API/Concepts/apisigningkey.htm)
+- Terraform 0.10.x+
+
+
+### Usage
+
+- Update `terraform.tfvars` with the required information. Terraform automatically loads them to populate variables, but you can also use the -var-file flag directly to specify a file. These files are the same syntax as Terraform configuration files. And like Terraform configuration files, these files can also be JSON.
+
+We don't recommend saving usernames and passwords to version control, but you can create a local secret variables file and use -var-file to load it.
+
+Most examples use the same set of environment variables so you only need to do this once. There's a sample file available on this sample code for your reference in addition to the snippet below:
+
+```
+###############################
+#### Environment variables ####
+###############################
+tenancy_ocid="<tenancy OCID>"
+compartment_ocid="<compartment OCID>"
+user_ocid="<tenancy OCID>"
+fingerprint="<PEM key fingerprint>"
+private_key_path="<RSA PEM key which matches with the PEM key fingerprint>"
+ssh_private_key_path="<path to the ssh private key to ssh to the instance>"
+ssh_public_key_path="<path to the ssh public key to setup on the KVM Host>"
+region="<OCI region>"
+
+#######################################
+#### Instance definition variables ####
+#######################################
+
+#Customer Identifier to be used before the resources name
+customer_name = "mycustomer"
+
+#availability_domain number - For AD1 uses 1. For AD2, uses 2, For AD3, uses 3
+availability_domain = "1"
+
+#Oracle Linux 7.4 Image OCID. Image OCI for specific region is available in the documentation:
+# https://docs.us-phoenix-1.oraclecloud.com/Content/Compute/References/images.htm
+variable "image_ocid" {
```

Comment:
Recommend to make this a map with region as the key:

variable "image_ocid" {
  type = "map"
  default = {
    // Oracle-provided image "Oracle-Linux-7.4-2017.12.18-0"
    // See https://docs.us-phoenix-1.oraclecloud.com/Content/Resources/Assets/OracleProvidedImageOCIDs.pdf
    us-phoenix-1 = "ocid1.image.oc1.phx.aaaaaaaasc56hnpnx7swoyd2fw5gyvbn3kcdmqc2guiiuvnztl2erth62xnq"
    us-ashburn-1 = "ocid1.image.oc1.iad.aaaaaaaaxrqeombwty6jyqgk3fraczdd63bv66xgfsqka4ktr7c57awr3p5a"
    eu-frankfurt-1 = "ocid1.image.oc1.eu-frankfurt-1.aaaaaaaayxmzu6n5hsntq4wlffpb4h6qh6z3uskpbm5v3v4egqlqvwicfbyq"
  }
}